### PR TITLE
fix cache quest lock condition

### PR DESCRIPTION
**Pull request by self-transition**
In the previous PR https://github.com/webzard-io/k8s-api-provider/pull/34, we added a synchronization lock to `globalStore.get()` to ensure that only one valid http(s) and wbsocket(s) request is sent for the same resource concurrently, but the lock is not sufficiently restrictive: the lock is only added if the request parameters are identical, for the following reason:
- The parameters passed to the dataprovider by the caller will only be used inside the dataprovider and not sent to the k8s api server, the actual request parameters sent to the api server should be determined by the dataprovider.
- In one globalstore instance, we should strictly ensure that only one list watch exists for a resource at a time, otherwise there will be unintended consequences.

This PR also adds the cancelquery logic that was missing: when the websocket readyState is CONNECTING, the connection should also be closed.

### CODE REVIEW

File: src/global-store.ts [hunk 1]
```diff
@@ -74,7 +73,11 @@ export class GlobalStore {
     }
 
     const cacheRequest = this.requestsCache.find(f => {
-      return f.resource === resource && isEqual(f.meta, meta);
+      // TODO: Its supposed to be strictly comparison of query path
+      return (
+        f.resource === resource &&
+        f.meta?.resourceBasePath === meta?.resourceBasePath
```
Comment (huangjunjia) [inline src/global-store.ts:79]: Why only compare the resourceBasePath and not the f.meta?

Comment (self-transition) [reply src/global-store.ts:79]: As mentioned in the PR description: the meta attribute contains request parameters, which should be ignored.

Comment (huangjunjia) [reply src/global-store.ts:79]: ok.

**Review by huangjunjia — APPROVED** [review]
LGTM

**Review by tanbowensg — APPROVED** [review]
LGTM

**Pull request merged**